doc/specs/wiki.md

```diff
@@ -0,0 +1,348 @@
+
+**Primary Actor:** A user
+
+**Brief:** Sometimes wiki comments need to be deleted.
+
+**Precondition:** The user has the wiki management permission or the wiki administration permission enabled.
+
+**Trigger:** The manager clicks the delete button above a comment.
+
+**Postcondition:** The comment is deleted.
+
+**Basic Flow:**
+
+1. The manager clicks the delete button above a comment.
+2. The manager is prompted with a confirmation window, ensuring they actually want to delete the comment.
+3. The comment is deleted if the manager clicks yes.
+
+**Functional Requirements:**
+
+* The corresponding comment's document must be deleted from the database.
+
+### 7.0 Tagging articles
+
+**Primary Actor:** A user
+
+**Brief:** Similar to the aliasing system, tagging allows users to search for articles on a broader topic.
+
+**Precondition:** The user has the appropriate user class to edit the article, or has the wiki administration permission enabled.
+
+**Trigger:** The user saves a tag on an article.
+
+**Postcondition:** The tag is saved.
+
+**Basic Flow:**
+
+1. The user types in the tag to add.
+2. The user hits submit, and the tag is saved.
+
+**Functional Requirements:**
+
+* Duplicate or blank tags must be ignored.
+* A new revision must be created when a tag is added.
+
+### 7.1 Removing article tags
+
+**Primary Actor:** A user
+
+**Brief:** Tags must be able to be removed for accuracy.
+
+**Precondition:** The user has the appropriate user class to edit the article and has the wiki management permission enabled, or has the wiki administration permission enabled.
+
+**Trigger:** The user clicks the X button next to a tag.
+
+**Postcondition:** The tag is removed.
+
+**Basic Flow:**
+
+1. The user clicks the X button next to a tag.
+2. The tag is deleted.
+
+**Functional Requirements:**
+
+* A new revision must be created when a tag is removed.
+
+### 8.0 Managing an article's dedicated terms
+
+**Primary Actor:** A user
+
+**Brief:** Dedicated terms allow common searches to resolve to the correct article.
+
+**Precondition:** The user has the appropriate class to edit the article and has the wiki management permission enabled, or has the wiki administration permission enabled.
+
+**Trigger:** The user adds or removes a dedicated term from an article.
+
+**Postcondition:**  N/A
+
+**Basic Flow:**
+
+1. The user adds or removes a dedicated term from an article.
+2. The update is saved.
+
+**Applicable business rules:**
+
+* Dedicated terms must be unique across all articles.
+
+## Data model
+
+**Wiki Articles:**
+* The title of the article.
+* The ID of the user that created the article.
+* An array of unique dedicated terms.
+* The timestamp of the article creation time.
+* The timestamp of the last edit time.
+* A sections subdocument, containing:
+  * The title of the section.
+  * The body of the section.
+  * A sub-section document, containing:
+    * The title of the sub-section.
+    * The body of the sub-section.
```

Comment:
Aren't sections just part of the bbcode? Should we really split those into separate documents?